### PR TITLE
Runtime extension types

### PR DESCRIPTION
**Pull request by hosted-fornet**

### CODE REVIEW

File: src/kernel_types.rs [hunk 1]
```diff
@@ -392,3 +392,22 @@ pub fn en_wit_send_error_kind(kind: SendErrorKind) -> wit::SendErrorKind {
...
+pub enum PythonResponse {
+    Run,
+    Err(String),
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum MessageType {
+    Request,
+    Response,
+}
```
Comment (dr-frmr) [inline src/kernel_types.rs:413]: The point of python executer being a runtime extension is so we don't have to put its types and API into the kernel, so we need to remove these and keep them somewhere else. Perhaps in a library shared by the runtime ext repo. Anywhere but here!

**Review by dr-frmr — CHANGES_REQUESTED** [review]

File: src/http.rs [hunk 2]
```diff
@@ -375,7 +401,12 @@ where
 
 /// Register a WebSockets path with the HTTP server. Your app must do this
 /// in order to receive incoming WebSocket connections.
-pub fn bind_ws_path<T>(path: T, authenticated: bool, encrypted: bool) -> anyhow::Result<()>
+pub fn bind_ws_path<T>(
+    path: T,
+    authenticated: bool,
+    encrypted: bool,
+    extension: bool,
+) -> anyhow::Result<()>
```
Comment (dr-frmr) [inline src/http.rs:409]: In this case I'd prefer to keep the main bind function with the old signature, and make a new function called `bind_extension_path` or something like that.

File: src/http.rs [hunk 3]
```diff
@@ -406,6 +432,39 @@ where
...
+        return Err(HttpServerError::PathBindError {
+            error: "http_server gave unexpected response".to_string(),
+        });
+    };
+    resp
+}
+
+/// Register a WebSockets path with the HTTP server specifically for sending and
+/// receiving system messages from a runtime extension. Only use this if you are
+/// writing a runtime extension.
```
Comment (hosted-fornet) [inline src/http.rs:456]: This looks good but I'd change the comment slightly, since the WS for the ext can also be used to serve a FE: its both.

**Review by hosted-fornet — COMMENTED** [review]
Comment on comment

**Review by dr-frmr — APPROVED** [review]

**Pull request merged**